go-controller/pkg/ovn/gateway.go

```diff
@@ -382,8 +404,16 @@ func (gw *GatewayManager) GatewayInit(
 
 	gwLRPMAC := util.IPAddrToHWAddr(gwLRPIPs[0])
 	gwLRPNetworks := []string{}
-	for _, gwLRPIfAddr := range gwLRPIfAddrs {
-		gwLRPNetworks = append(gwLRPNetworks, gwLRPIfAddr.String())
+	for _, gwLRPJoinIP := range gwLRPJoinIPs {
+		gwLRPNetworks = append(gwLRPNetworks, gwLRPJoinIP.String())
+	}
+	if gw.netInfo.TopologyType() == types.Layer2Topology {
+		// At layer2 GR LRP acts as the layer3 ovn_cluster_router so we need
+		// to configure here the .1 address, this will work only for IC with
+		// one node per zone, since ARPs for .1 will not go beyond local switch.
+		for _, subnet := range hostSubnets {
+			gwLRPNetworks = append(gwLRPNetworks, util.GetNodeGatewayIfAddr(subnet).String())
```

Comment:
I'm not sure why this is needed either. If I launch a origin/master cluster I see it has .1 address already on it:

```
### no patch
router 672a80e9-932e-4679-925e-3578a4918a09 (GR_default.l2.primary_ovn-control-plane)
    port rtos-default.l2.primary_ovn_layer2_switch
        mac: "0a:58:64:41:00:02"
        ipv6-lla: "fe80::858:64ff:fe41:2"
        networks: ["10.20.0.1/16", "100.65.0.2/16"]
    port rtoe-GR_default.l2.primary_ovn-control-plane
        mac: "02:42:ac:12:00:03"
        ipv6-lla: "fe80::42:acff:fe12:3"     
        networks: ["169.254.0.11/17", "172.18.0.3/16"]

### with this patch

    port rtoe-GR_default.l2.primary_ovn-control-plane
        mac: "02:42:ac:12:00:03"
        ipv6-lla: "fe80::42:acff:fe12:3"
        networks: ["169.254.0.11/17", "172.18.0.3/16"]
    port rtos-default.l2.primary_ovn_layer2_switch
        mac: "0a:58:64:41:00:02"
        ipv6-lla: "fe80::858:64ff:fe41:2"
        networks: ["10.20.0.1/16", "100.65.0.2/16"]

```

Reply:
oh I see you removed it from https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4819/commits/dca22499b961a2ee0f03e524e43e82e767b79f7c#diff-cbf27b8cde9a56e399efde996bf12bc642fee29aae6c0bd8236a42105880ebdbL722-L728